oci/common.go

```diff
@@ -136,56 +136,19 @@ func (o *OrasRemote) setRepository(ref registry.Reference) error {
 		ref.Registry = "ghcr.io"
 		ref.Repository = "defenseunicorns/packages/" + ref.Repository
 	}
-	client, err := o.createAuthClient(ref)
+	storeOpts := credentials.StoreOptions{}
+	credStore, err := credentials.NewStoreFromDocker(storeOpts)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to get credentials: %w", err)
+	}
+	client := &auth.Client{
+		Client:     retry.DefaultClient,
+		Cache:      auth.NewCache(),
+		Credential: credentials.Credential(credStore),
 	}
 
 	o.repo.Reference = ref
 	o.repo.Client = client
 
 	return nil
 }
-
-// createAuthClient returns an auth client for the given reference.
-//
-// The credentials are pulled using Docker's default credential store.
-//
-// TODO: instead of using Docker's cred store, should use the new one from ORAS to remove that dep
-func (o *OrasRemote) createAuthClient(ref registry.Reference) (*auth.Client, error) {
-
-	client := o.repo.Client.(*auth.Client)
-	o.log.Debug(fmt.Sprintf("Loading docker config file from default config location: %s for %s", config.Dir(), ref))
-	cfg, err := config.Load(config.Dir())
-	if err != nil {
-		return nil, err
-	}
-	if !cfg.ContainsAuth() {
-		o.log.Debug("no docker config file found")
-		return client, nil
-	}
-
-	configs := []*configfile.ConfigFile{cfg}
-
-	var key = ref.Registry
-	if key == "registry-1.docker.io" {
-		// Docker stores its credentials under the following key, otherwise credentials use the registry URL
-		key = "https://index.docker.io/v1/"
-	}
```

Comment:
This behavior is _slightly_ changed now https://github.com/oras-project/oras-go/blob/d51a392ff5432a9090c64ffec6ca6a8690b55e18/registry/remote/credentials/registry.go#L87-#L89 but I believe does not break anything